enhancements/machine-api/control-plane-set.md

```diff
@@ -0,0 +1,585 @@
+changes that will cause version skew issues over the next handful of releases.
+
+### Operational Aspects of API Extensions
+
+We will introduce a new ClusterOperator for the ControlPlaneSetOperator. Within this we will introduce conditions (TBD)
+which describe the state of the operator and the control plane which it manages.
+
+This new ClusterOperator will be the key resource from which support should discover information if they believe the
+ControlPlaneSet to be degraded.
+
+There should be no effect to existing operators or supportability of other components due to this enhancement, in
+particular, this is strictly adding new functionality on top of an existing API, we do not expect it to impact the
+functionality of other APIs.
+
+#### Failure Modes
+
+- A new Machine fails to scale up
+  - In this scenario, we expect the operator to turn degraded to signal that there is an issue with the Control Plane
+    Machines
+  - We expect that the Machine will contain information identifying the issue with the launch request, which can be
+    rectified by the user before they then delete the Machine, once deleted, the ControlPlaneSet will attempt the
+    scale up again
+  - This process should be familiar from dealing with existing Failed Machines
+  - A MachineHealthCheck targeting ControlPlaneMachines could automatically resolve this issue
+- The ControlPlaneSet webhook is not operational
+  - This will prevent creation and update of ControlPlaneSets
+  - We expect these operations to be very infrequent within a cluster once established, so the impact should be
+    minimal
+  - The failure policy will be to fail closed, as such, when the webhook is down, no update operations will succeed
+  - The Kube API server operator already detects failing webhooks and as such will identify the issue early
+
+#### Support Procedures
+
+TBD
+
+## Implementation History
+
+There is not current implementation of this enhancement, however, it will depend on the
+[etcd quorum protection mechanism](https://github.com/openshift/enhancements/pull/943) which is currently being
+implemented.
+
+## Drawbacks
+
+- Introduction of new CRDs complicates the user experience
+  - Users will have to learn about the new CRD and how it can be used
+  - Counter: The concepts are familiar from existing apps resources and CRDs
+- We are making it easier for customers to put their clusters at risk
+  - If a scale up fails, this will then need manual intervention, the cluster will become degraded at this point
+  - Counter: The etcd protection mechanism should mean that the cluster is never at risk of losing quorum, clusters
+    should be recoverable
+- We will need confidence in the project before we can ship it
+  - We need to decide how to ship the project, will we ship it as GA or TP in the first instance?
+
+## Alternatives
+
+Previous iterations of this enhancement linked in the metadata describe alternative ways we could implement this
+feature.
```

Comment:
Given our general desire to stop adding new things to the release payload, and the fact that this new feature seems to be opt-in, did you explore delivering this new operator via OLM instead of the payload?

Reply:
I was imagining that this would eventually be (4.12) default and set up by the installer, so I hadn't investigated the idea of OLM. If we do want to make this opt-in permanently then having it delivered via OLM could be an option.

When I've been talking with the etcd team about this in the past, we had imagined they would also observe the CPS resource to understand control plane scale in the longer term as well as they don't have a good way to see how many replicas they should have at the moment

Reply:
I'm going to add an open question for is this optional or not